solr/core/src/test/org/apache/solr/search/join/TestCloudNestedDocsSort.java

```diff
@@ -131,7 +131,7 @@ public static void setupCluster() throws Exception {
 
   @AfterClass
   public static void cleanUpAfterClass() {
-    client = null;
```

Comment:
In a number of places, you are changing code that nulled static fields to not do so any more; only close the resource.  But I think there is a retained memory/heap concern?  Maybe if we look carefully at the test run performance, we'll see if this is an actual concern or not.  Running with only a couple threads would better ascertain this (vs more).